backend/tarsy/controllers/history_controller.py

```
@@ -225,6 +233,37 @@ async def get_session_detail(
+        # Process chain execution data if available
+        chain_execution = None
+        if chain_execution_data and session_info.get('chain_id'):
+            stages_data = chain_execution_data.get('stages', [])
+            stage_executions = []
+            
+            for stage_data in stages_data:
+                stage_execution = StageExecution(
+                    execution_id=stage_data.get('execution_id', ''),
+                    stage_id=stage_data.get('stage_id', ''),
+                    stage_index=stage_data.get('stage_index', 0),
+                    stage_name=stage_data.get('stage_id', '').split('_')[0] if stage_data.get('stage_id') else '',  # Extract name from stage_id
```

Ruff (0.12.2)

Line too long (144 > 88)

(E501)